src/main/java/pl/betoncraft/betonquest/compatibility/citizens/CitizensParticle.java

```diff
@@ -75,22 +82,81 @@ public void run() {
 		enabled = true;
 	}
 
+
 	@Override
 	public void run() {
-		for (NPC npc : npcs) {
-			if (npc == null) {
-				continue;
-			}
-			Entity e = npc.getEntity();
-			if (e == null) {
-				continue;
-			}
-			Location loc = e.getLocation().clone();
-			loc.setPitch(-90);
-			Compatibility.getEffectManager().start(name, section, loc);
-		}
+	    // Bukkit.getLogger().info("Particle: RUN:89: Online Players: " + Bukkit.getOnlinePlayers().size() + "NPC:" + npcs.size());
+        for (NPC npc : npcs) {
+            if (npc == null) {
+                continue;
+            }
+            Entity e = npc.getEntity();
+            if (e == null) {
+                continue;
+            }
+            // Bukkit.getLogger().info("NPC Name: " + npc.getName());
+            Location loc = e.getLocation().clone();
+            loc.setPitch(-90);
+            // Compatibility.getEffectManager().start(name, section, loc);
```

Comment:
You're right, this checks' (in the constructor) outcome should be stored in a boolean field:

		if (section == null)
			return;
		if ("true".equalsIgnoreCase(section.getString("disabled")))
			return;
		name = section.getString("class");
		if (name == null)
			return;

And the loop at line 89 should run only if it's `true`. No need to make that checks every time it runs.